src/cli/cmd/login/mod.rs

```diff
@@ -288,6 +374,7 @@ async fn upsert_user_nix_config(
         if update_nix_conf {
             nix_conf_write_success = match tokio::fs::OpenOptions::new()
                 .create(true)
+                .truncate(true)
```

Comment:
Strong -1 on this. We should not completely overwrite users' `nix.conf` files. That's why we have this "upsert" functionality below, roughly based on nix-installer's -- so we only add / modify the settings we care about, and (mostly) don't touch anything else.

Reply:
Well, we do read the file to update it "in place" (unless I've missed a critical detail) but we do so from the same file path, with a different file handle.

It feels weird to open a handle, not truncate, then read from a different handle, then "write_all" to the first. What does `write_all` do, for example, if the written bytes are smaller than the existing file? The docs weren't clear.

Reply:
okay, I just switched it back to not truncating so the read+merge will work as before.

Reply:
OK, I made a dumb. Re-reading the code, I think it'd be fine to truncate it, because we'll be rewriting it with our combined upserted settings and their original settings, I think.....

But I'd like to see that that's actually what happens (maybe a test or something -- or you running it and saying "yeah that's what happens", either-or)

Reply:
pushed another separate commit that reorders this, please review specifically.

Reply:
Reflects the logic I'd expect to see :+1: